src/decisionengine/framework/engine/SourceWorkers.py

```diff
@@ -174,29 +174,45 @@ def update(self, channel_name, source_configs):
                 worker = SourceWorker(key, config, channel_name, self._exchange, self._broker_url)
                 self._workers[key] = worker
                 workers[key] = worker
+                self._use_count[key] = {channel_name}
 
         return workers
 
-    def prune(self, source_names):
+    def unique(self, source_name):
```

Comment:
maybe calling this `is_one()`?

Reply:
I'll stick with `unique()` for now, but I agree a better term could be used.